core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1002,6 +1001,7 @@ class DAGScheduler(
     val stageId = task.stageId
     val taskType = Utils.getFormattedClassName(task)
 
+    // REVIEWERS: does this need special handling for multiple completions of the same task?
```

Comment:
great, thanks mark.  I'll leave the comment in for now just to get any other reviewers to take a look too.  Think this deserve some special handling in the test cases?  (and if so, any suggestions / pointers on what to do?)  I'm not very familiar w/ the OutputCommitCoordinator so guidance is appreciated.